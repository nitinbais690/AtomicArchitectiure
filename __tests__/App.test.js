/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import { render, waitFor } from '@testing-library/react-native';
import SeriesScreen from "@series/screens/series-screen"

jest.mock('axios')

describe('<SeriesScreen />', () => {

  it('renders series screen correctly', async () => {
    const tree = render(<SeriesScreen />);
    await waitFor(() => {
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

});
