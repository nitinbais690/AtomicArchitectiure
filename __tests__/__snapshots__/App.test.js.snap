// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeriesScreen /> renders series screen correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    horizontal={false}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": 1334,
            "resizeMode": "cover",
            "width": "100%",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://resizer-staging-tvpass.azureedge.net/image/A604A743-3A73-4A3D-B02B-839FE0965055/0-16x9.jpg?width=1280",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 1334,
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <BVLinearGradient
          colors={
            Array [
              2567704609,
              449366216,
            ]
          }
          endPoint={
            Object {
              "x": 0.5,
              "y": 0.5,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0.3,
              "y": 0.5,
            }
          }
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "flex-start",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "height": "100%",
                  "paddingTop": "2%",
                  "width": "8%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#686EFF",
                    "borderBottomRightRadius": 66.5,
                    "borderTopRightRadius": 66.5,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingLeft": 5.5,
                    "paddingRight": 8,
                    "paddingVertical": 33.5,
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/credits.png",
                    }
                  }
                  style={
                    Object {
                      "height": 33.5,
                      "marginRight": 8,
                      "width": 33.5,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Inter-Regular",
                      "fontSize": 33.5,
                      "lineHeight": 34.5,
                    }
                  }
                >
                  1250
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "height": "100%",
                  "marginRight": "8%",
                  "width": "80%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexDirection": "column",
                    "height": "85%",
                    "justifyContent": "flex-end",
                    "paddingBottom": "3%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "flexDirection": "column",
                          "paddingBottom": 53.5,
                        }
                      }
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../../assets/amc_logo.png",
                          }
                        }
                        style={
                          Object {
                            "height": 69.5,
                            "marginBottom": 29.5,
                            "width": 120,
                          }
                        }
                      />
                      <Text
                        style={
                          Object {
                            "color": "#FFFFFF",
                            "fontFamily": "Inter-Medium",
                            "fontSize": 41.5,
                            "lineHeight": 45.5,
                          }
                        }
                      >
                        Ghost Dimension
                      </Text>
                      <View
                        style={
                          Object {
                            "paddingBottom": 18.5,
                            "paddingTop": 18.5,
                          }
                        }
                      >
                        <Text
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "#FFFFFF",
                              "fontFamily": "Inter-SemiBold",
                              "fontSize": 100,
                              "lineHeight": 113.5,
                            }
                          }
                          testID="currentEpisodeTitle"
                        >
                          S2 E1: A Poltergeist Nightmare at 30 East Drive
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "paddingBottom": 26.5,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#9BADBE",
                              "fontFamily": "Inter-Regular",
                              "fontSize": 33.5,
                              "lineHeight": 40,
                            }
                          }
                        >
                          2017 - Horror, Documentary - MA - 53M
                        </Text>
                      </View>
                      <Text
                        style={
                          Object {
                            "color": "#FFFFFF",
                            "fontFamily": "Inter-Medium",
                            "fontSize": 41.5,
                            "lineHeight": 45.5,
                          }
                        }
                      >
                        The team step into a house that has become legend with the paranormal community, 30 East Drive, Pontefract.
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        hasTVPreferredFocus={false}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                        testID="reedemButton"
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#686EFF",
                              "borderColor": "#FFFFFF",
                              "borderRadius": 66.5,
                              "borderWidth": 2,
                              "flexDirection": "row",
                              "fontFamily": "Inter-Regular",
                              "justifyContent": "center",
                              "paddingHorizontal": 66.5,
                              "paddingVertical": 28,
                              "width": "100%",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/credits.png",
                              }
                            }
                            style={
                              Object {
                                "height": 40,
                                "marginRight": 8,
                                "width": 40,
                              }
                            }
                          />
                          <Text
                            style={
                              Object {
                                "color": "#FFFFFF",
                                "fontFamily": "Inter-Medium",
                                "fontSize": 40,
                                "lineHeight": 44,
                              }
                            }
                          >
                            Watch for 10 credits
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </BVLinearGradient>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0C1021",
            "height": "100%",
            "justifyContent": "center",
            "paddingHorizontal": "8%",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "paddingTop": 66.5,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#9BADBE",
                  "fontFamily": "Inter-Medium",
                  "fontSize": 41.5,
                  "lineHeight": 45.5,
                }
              }
            >
              Seasons & Episodes
            </Text>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "backgroundColor": "#9BADBE",
                  "height": 4.5,
                  "justifyContent": "center",
                  "marginBottom": 36,
                  "marginTop": 12,
                  "opacity": 0.5,
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "height": "100%",
                  "width": "19.4%",
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "column",
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <RCTScrollView
                  data={
                    Array [
                      SeasonModel {
                        "episodes": Array [
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "The team step into a house that has become legend with the paranormal community, 30 East Drive, Pontefract. With a blockbuster movie behind it, the team are about to find out that even after 50 years since the Pritchard family underwent their horrific ordeal, there is still a darkness that pervades this ordinary looking house on an ordinary street.",
                            "duration": 3180,
                            "episodeNo": 1,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "0B8E168A-674A-49E9-A4E9-3AC985D486F3",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "The team step into a house that has become legend with the paranormal community, 30 East Drive, Pontefract.",
                            "status": "published",
                            "title": "A Poltergeist Nightmare at 30 East Drive",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "The team head to Lincolnshire to investigate the Village Farm Museum. This location looks out of place being in the middle of a residential housing estate. The original farmhouse has its fair share of ghostly goings on and for one of the team it seems they may have attracted a new friend.",
                            "duration": 2700,
                            "episodeNo": 2,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "1490A41C-E123-483D-804B-C7C21722D760",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "The team head to Lincolnshire to investigate the Village Farm Museum. This location looks out of place being in the middle of a residential housing estate.",
                            "status": "published",
                            "title": "Sinister Happenings at The Village Farm Museum Part 1",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "The team head to Lincolnshire to investigate the Village Farm Museum. This location looks out of place being in the middle of a residential housing estate. The original farmhouse has its fair share of ghostly goings on and for one of the team it seems they may have attracted a new friend.",
                            "duration": 2700,
                            "episodeNo": 3,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "B8EE90B4-BFE2-4702-8208-AEFFA323789E",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "The team head to Lincolnshire to investigate the Village Farm Museum. This location looks out of place being in the middle of a residential housing estate.",
                            "status": "published",
                            "title": "Sinister Happenings at The Village Farm Museum Part 2",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "The team encounter a spirit member joining them on the thermal imaging camera and appears to leave its ghostly hand print on camera. A ghostly dog makes itself known during an interview with the psychic medium and a spirit seems to vanish some of the teams equipment, never to be seen again.",
                            "duration": 2700,
                            "episodeNo": 4,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "63BDACD4-F6AB-4C60-9176-54B7BFBC54ED",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "The team encounter a spirit member joining them on the thermal imaging camera and appears to leave its ghostly hand print on camera.",
                            "status": "published",
                            "title": "Souls Seeking Solace Gotts Park Mansion",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "The ghost dimension team visit Batley, West Yorkshire to investigate a location where it may not be the building that's haunted but the items that reside within. Currently used an an auction house, could it be a haunted object amongst the many hundreds that pass through here or is a former owner/employee of the building reluctant to move on even after the hammer has gone down.",
                            "duration": 2700,
                            "episodeNo": 5,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "CC94A493-BD0D-48A3-BEE1-76FD2CC7F79D",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "The ghost dimension team visit Batley, West Yorkshire to investigate a location where it may not be the building that's haunted but the items that reside within.",
                            "status": "published",
                            "title": "Long Lost Treasure at Jessops Old Mill and Auction House",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "Four floors of terror. Objects from around the world are at home at the museum but what has attached themselves to them and now roam the floors and corridors after dark? Is there a little boy that knows how to set off a hand drier in a rest room? And can an animal die from a broken heart? Prepare for a night of terror.",
                            "duration": 2700,
                            "episodeNo": 6,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "E1E55E85-E476-414B-8ABF-5CFF01D21320",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "Four floors of terror. Objects from around the world are at home at the museum but what has attached themselves to them and now roam the floors and corridors after dark.",
                            "status": "published",
                            "title": "A Night at The Leeds Museum Part 1",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "With activity sending the team into a spin on the top floor, is there something attached to the museums main attraction, the mummy? Is there a curse with it or have the team just caught sight and captured on film the real culprit that haunts the museum, a lady called Jenny?",
                            "duration": 2700,
                            "episodeNo": 7,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "6B01D343-EC19-4D40-B7EA-2AD91B10975C",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "With activity sending the team into a spin on the top floor, is there something attached to the museums main attraction, the mummy.",
                            "status": "published",
                            "title": "A Night at The Leeds Museum Part 2",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "The team head to Liverpool to help a restaurant uncover what could be haunting here. Bex and Sean encounter the spirit of a white lady in the private quarters. A spirit interacts with Bex in the kitchen and Sean speaks to lost souls in an unforgettable moment.",
                            "duration": 2640,
                            "episodeNo": 8,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "6C58482E-3D57-4D37-B788-13BD0F0ECEF3",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "The team head to Liverpool to help a restaurant uncover what could be haunting here. Bex and Sean encounter the spirit of a white lady in the private quarters.",
                            "status": "published",
                            "title": "Lost Souls of One Melling Road",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "Bex and Sean head into Northwich, Cheshire to investigate a disused bingo hall. Bex braves a loan vigil, which she will never forget as a door seemingly opens on its own, objects are moved and a spirit takes a seat with Sean on his investigation. Are the spirits still waiting for their final numbers to be called?",
                            "duration": 2700,
                            "episodeNo": 9,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "AF25A750-3F45-4026-A4AE-46C24BFBEE2C",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "Bex and Sean head into Northwich, Cheshire to investigate a disused bingo hall.",
                            "status": "published",
                            "title": "Full House at Northwich Plaza and Bingo Hall",
                            "year": 2017,
                          },
                          EpisodeModel {
                            "airedDate": "2017-01-01T01:00:00Z",
                            "creditPoints": 3,
                            "description": "The team head to Elland to investigate ghostly goings on in this haunted pub. Its not just the spirits behind the bar lurking here. Multiple hauntings, murders and death surround this public house. Will headless horses make an appearance tonight and who or what haunts the beer cellar?",
                            "duration": 2640,
                            "episodeNo": 10,
                            "genres": Array [
                              "Horror",
                              "Documentary",
                            ],
                            "id": "E52819C3-0343-406B-95EE-AE2FF025C82C",
                            "image": "https://resizer-staging-tvpass.azureedge.net/image/0B8E168A-674A-49E9-A4E9-3AC985D486F3/0-16x9.jpg?width=276",
                            "seasonNo": 2,
                            "shortDescription": "The team head to Elland to investigate ghostly goings on in this haunted pub. Its not just the spirits behind the bar lurking here. Multiple hauntings, murders and death surround this public house.",
                            "status": "published",
                            "title": "Last Orders at The Fleece Inn",
                            "year": 2017,
                          },
                        ],
                        "id": "629D69EF-BC97-41F3-8733-1513528C7416",
                        "image": "https://resizer-staging-tvpass.azureedge.net/image/A604A743-3A73-4A3D-B02B-839FE0965055/0-16x9.jpg?width=1280",
                        "number": 2,
                        "seriesTitle": "Ghost Dimension",
                        "title": "Ghost Dimension Season 2",
                        "totalCreditPoints": "",
                      },
                    ]
                  }
                  disableVirtualization={false}
                  getItem={[Function]}
                  getItemCount={[Function]}
                  horizontal={false}
                  initialNumToRender={10}
                  keyExtractor={[Function]}
                  maxToRenderPerBatch={10}
                  nestedScrollEnabled={true}
                  onContentSizeChange={[Function]}
                  onEndReachedThreshold={2}
                  onLayout={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  removeClippedSubviews={false}
                  renderItem={[Function]}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={Array []}
                  updateCellsBatchingPeriod={50}
                  viewabilityConfigCallbackPairs={Array []}
                  windowSize={21}
                >
                  <View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        hasTVPreferredFocus={false}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                        testID="seasonItem"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "borderColor": "white",
                                "borderRadius": 66.5,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "opacity": 1,
                                "paddingVertical": 26.5,
                                "width": "100%",
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#9BADBE",
                                "fontFamily": "Inter-Medium",
                                "fontSize": 41.5,
                                "lineHeight": 45.5,
                              }
                            }
                          >
                            Season 2
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "height": "100%",
                  "width": "78%",
                }
              }
            >
              <View />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
